src/main/kotlin/org/jitsi/nlj/transform/node/incoming/LayerLookup.kt

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright @ 2018 - Present, 8x8 Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jitsi.nlj.transform.node.incoming
+
+import org.jitsi.nlj.Event
+import org.jitsi.nlj.PacketInfo
+import org.jitsi.nlj.SetMediaSourcesEvent
+import org.jitsi.nlj.rtp.VideoRtpPacket
+import org.jitsi.nlj.stats.NodeStatsBlock
+import org.jitsi.nlj.transform.node.TransformerNode
+import org.jitsi.utils.logging2.Logger
+import org.jitsi.utils.logging2.createChildLogger
+import org.jitsi.nlj.MediaSourceDesc
+import org.jitsi.nlj.RtpLayerDesc
+import java.util.concurrent.atomic.AtomicInteger
+
+/**
+ * Set video packets' quality layer info
+ */
+class LayerLookup(
+    parentLogger: Logger
+) : TransformerNode("Video parser") {
+    private val logger = createChildLogger(parentLogger)
+    private var sources: Array<MediaSourceDesc> = arrayOf()
+    private val numPacketsDroppedNoEncoding = AtomicInteger()
+
+    /* TODO: combine this with VideoBitrateCalculator? They both do findRtpLayerDesc. */
+    override fun transform(packetInfo: PacketInfo): PacketInfo? {
+        val videoPacket = packetInfo.packetAs<VideoRtpPacket>()
+        val encodingDesc = findRtpLayerDesc(videoPacket) ?: run {
+            logger.warn("Unable to find encoding matching packet! packet=$videoPacket; " +
+                "encodings=${sources.joinToString(separator = "\n", limit = 1,
+                    truncated = "[${sources.size - 1} more source descriptions omitted]")}")
+            numPacketsDroppedNoEncoding.incrementAndGet()
+            return null
+        }
+        videoPacket.qualityIndex = encodingDesc.index
+
+        return packetInfo
+    }
+
+    private fun findRtpLayerDesc(packet: VideoRtpPacket): RtpLayerDesc? {
+        for (source in sources) {
```

Comment:
I think we have this in a few places, I wonder if it's worth defining `fun Array<MediaSourceDesc>.findRtpLayerDesc` ?